src/routes/cache.ts

```diff
@@ -15,4 +16,18 @@ cacheRouter.get(/^\/origin\/[a-zA-Z0-9]+\/[0-9]+\/H\.Cache\.bin.*$/, (_req, res)
     res.sendFile(`static/data/H.Cache_${config.version}.bin`, { root: "./" });
 });
 
+cacheRouter.get(/\.bk2!/, async (req, res) => {
+    const dir = req.path.substr(0, req.path.lastIndexOf("/"));
+    const file = req.path.substr(dir.length + 1);
+
+    // Return file if we have it
+    if (fs.existsSync(`static/data${dir}/${file}`)) {
+        res.send(fs.readFileSync(`static/data${dir}/${file}`, null));
```

Comment:
wouldn't readFile be better as it is non-blocking? same for the other Sync?

Reply:
You would know that better than me. From what I understand, JavaScript is single-threaded either way.

Reply:
based on my experience using non-sync methods is always better.
Yes, js is single-threaded however the whole server blocks while using a sync method, so lets use the non-sync ones here.

Reply:
I'm afraid I'm not sure how to elegantly write this code to be async. Plus, I'm not sure the overhead of reading from a modern disk is that significant. Might be worse we were dealing with multiple clients. I would use express' sendFile, but it errors due to the weird paths.

Reply:
ill check tomorrow!